src/raftstore/store/debug.rs

```diff
@@ -170,6 +187,157 @@ impl Debugger {
+                let write = box_try!(Write::parse(&value));
+                let mut write_info = WriteInfo::default();
+                write_info.set_start_ts(write.start_ts);
+                match write.write_type {
+                    WriteType::Put => write_info.set_field_type(Op::Put),
+                    WriteType::Delete => write_info.set_field_type(Op::Del),
+                    WriteType::Lock => write_info.set_field_type(Op::Lock),
+                    WriteType::Rollback => write_info.set_field_type(Op::Rollback),
+                }
+                let encoded_key = Key::from_encoded(keys::origin_key(&key).to_owned());
+                write_info.set_commit_ts(box_try!(encoded_key.decode_ts()));
+                writes.push(write_info);
+            }
+            return Ok(Some((prefix, RepeatedField::from_vec(writes))));
+        }
+        Ok(None)
+    }
+
+    fn next_grouped(iter: &mut DBIterator) -> Option<(Vec<u8>, Vec<Kv>)> {
+        if iter.valid() {
+            let prefix = keys::truncate_ts(iter.key()).to_vec();
+            let mut kvs = vec![(iter.key().to_vec(), iter.value().to_vec())];
+            while iter.next() && iter.key().starts_with(&prefix) {
+                kvs.push((iter.key().to_vec(), iter.value().to_vec()));
+            }
+            return Some((prefix, kvs));
+        }
+        None
+    }
+
+    fn next_item(&mut self) -> Result<Option<(Vec<u8>, MvccInfo)>> {
+        if self.limit != 0 && self.count >= self.limit {
+            return Ok(None);
+        }
+
+        let mut mvcc_info = MvccInfo::new();
+        let mut min_prefix = Vec::new();
+
+        let (lock_ok, writes_ok) = match (self.lock_iter.valid(), self.write_iter.valid()) {
+            (false, false) => return Ok(None),
+            (true, true) => {
+                let prefix1 = self.lock_iter.key();
+                let prefix2 = keys::truncate_ts(self.write_iter.key());
+                match prefix1.cmp(prefix2) {
+                    Ordering::Less => (true, false),
+                    Ordering::Equal => (true, true),
+                    _ => (false, true),
+                }
+            }
+            valid_pair => valid_pair,
+        };
+
+        if lock_ok {
+            if let Some((prefix, lock)) = self.next_lock()? {
+                mvcc_info.set_lock(lock);
+                min_prefix = prefix;
+            }
+        }
+        if writes_ok {
+            if let Some((prefix, writes)) = self.next_write()? {
+                mvcc_info.set_writes(writes);
+                min_prefix = prefix;
+            }
+        }
+        if self.default_iter.valid() {
+            match keys::truncate_ts(self.default_iter.key()).cmp(&min_prefix) {
+                Ordering::Equal => if let Some((_, values)) = self.next_default()? {
+                    mvcc_info.set_values(values);
+                },
+                Ordering::Greater => {}
+                _ => unreachable!(),
```

Comment:
If there is a bug, it's reachable. So should return all those keys instead of panic.